packages/create-llama/templates/python.ts

```diff
@@ -43,7 +50,29 @@ const addDependencies = async (
     const tool = fileParsed.tool as any;
     const existingDependencies = tool.poetry.dependencies as any;
     for (const dependency of dependencies) {
-      existingDependencies[dependency.name] = dependency.version;
+      const orignalValue = existingDependencies[dependency.name];
+      if (orignalValue === undefined) {
+        // Add dependency
+        if (dependency.version === undefined) {
+          throw Error(
+            `Dependency ${dependency.name} is missing version attribute!`,
+          );
+        } else {
+          existingDependencies[dependency.name] = dependency.version;
+        }
+      } else {
+        // Update dependency
+        if (typeof orignalValue === "string") {
+          existingDependencies[dependency.name] = {
+            version: dependency.version ?? orignalValue,
+            extras: dependency.extras,
+          };
+        } else {
+          orignalValue.version = dependency.version ?? orignalValue.version;
+          orignalValue.extras = dependency.extras;
+          existingDependencies[dependency.name] = orignalValue;
```

Comment:
i would generate a `value` and then always call this:
```suggestion
          existingDependencies[dependency.name] = value;
```

Reply:
this function is getting complicated - better factor it out, then we can easily unit test it (once we have unit tests for create-llama)